controllers/external/util.go

```diff
@@ -31,17 +31,17 @@ import (
 )
 
 // Get uses the client and reference to get an external, unstructured object.
-func Get(ctx context.Context, c client.Reader, ref *corev1.ObjectReference, namespace string) (*unstructured.Unstructured, error) {
+func Get(ctx context.Context, c client.Reader, ref *corev1.ObjectReference) (*unstructured.Unstructured, error) {
 	if ref == nil {
 		return nil, errors.Errorf("cannot get object - object reference not set")
 	}
 	obj := new(unstructured.Unstructured)
 	obj.SetAPIVersion(ref.APIVersion)
 	obj.SetKind(ref.Kind)
 	obj.SetName(ref.Name)
-	key := client.ObjectKey{Name: obj.GetName(), Namespace: namespace}
-	if err := c.Get(ctx, key, obj); err != nil {
-		return nil, errors.Wrapf(err, "failed to retrieve %s external object %q/%q", obj.GetKind(), key.Namespace, key.Name)
+	obj.SetNamespace(ref.Namespace)
+	if err := c.Get(ctx, client.ObjectKeyFromObject(obj), obj); err != nil {
+		return nil, errors.Wrapf(err, "failed to retrieve %s %s/%s", obj.GetKind(), ref.Namespace, ref.Name)
```

Comment:
Based on Alberto's comment. Let's go with
```go
		return nil, errors.Wrapf(err, "failed to retrieve %s external object %s", obj.GetKind(), klog.KRef(ref.Namespace, ref.Name))
```

Reply:
Actually used `KObj` there, since the unstructured is populated from ref. Looking into e2e CI failures, it seems some of the references are missing namespace.

Reply:
Interesting. I checked the webhook and I thought we should be good, but maybe not :)

Reply:
> Actually used KObj there, since the unstructured is populated from ref

I'm not sure if controller-runtime guarantees any specific content for an Unstructured if the Get call fails

Reply:
Yeah, that’s a valid concern. Updated it to KRef, KObj usage was due to misread comment.